python/samples/getting_started_with_agents/step4_kernel_function_strategies.py

```diff
@@ -0,0 +1,118 @@
+from semantic_kernel.kernel import Kernel
+
+REVIEWER_NAME = "ArtDirector"
+REVIEWER_INSTRUCTIONS = """
+You are an art director who has opinions about copywriting born of a love for David Ogilvy.
+The goal is to determine if the given copy is acceptable to print.
+If so, state that it is approved.
+If not, provide insight on how to refine suggested copy without example.
+"""
+
+COPYWRITER_NAME = "CopyWriter"
+COPYWRITER_INSTRUCTIONS = """
+You are a copywriter with ten years of experience and are known for brevity and a dry humor.
+The goal is to refine and decide on the single best copy as an expert in the field.
+Only provide a single proposal per response.
+You're laser focused on the goal at hand.
+Don't waste time with chit chat.
+Consider suggestions when refining an idea.
+"""
+
+
+def _create_kernel_with_chat_completion(service_id: str) -> Kernel:
+    kernel = Kernel()
+    kernel.add_service(AzureChatCompletion(service_id=service_id))
+    return kernel
+
+
+async def main():
+    agent_reviewer = ChatCompletionAgent(
+        service_id="artdirector",
+        kernel=_create_kernel_with_chat_completion("artdirector"),
+        name=REVIEWER_NAME,
+        instructions=REVIEWER_INSTRUCTIONS,
+    )
+
+    agent_writer = ChatCompletionAgent(
+        service_id="copywriter",
+        kernel=_create_kernel_with_chat_completion("copywriter"),
+        name=COPYWRITER_NAME,
+        instructions=COPYWRITER_INSTRUCTIONS,
+    )
+
+    termination_function = KernelFunctionFromPrompt(
+        function_name="termination",
+        prompt="""
+        Determine if the copy has been approved.  If so, respond with a single word: yes
+
+        History:
+        {{$history}}
+        """,
+    )
+
+    selection_function = KernelFunctionFromPrompt(
+        function_name="selection",
+        prompt=f"""
+        Determine which participant takes the next turn in a conversation based on the the most recent participant.
+        State only the name of the participant to take the next turn.
+        No participant should take more than one turn in a row.
+        
+        Choose only from these participants:
+        - {REVIEWER_NAME}
+        - {COPYWRITER_NAME}
+        
+        Always follow these rules when selecting the next participant:
+        - After user input, it is {COPYWRITER_NAME}'s turn.
+        - After {COPYWRITER_NAME} replies, it is {REVIEWER_NAME}'s turn.
+        - After {REVIEWER_NAME} provides feedback, it is {COPYWRITER_NAME}'s turn.
+
+        History:
+        {{{{$history}}}}
+        """,
+    )
+
+    chat = AgentGroupChat(
+        agents=[agent_writer, agent_reviewer],
+        termination_strategy=KernelFunctionTerminationStrategy(
+            agents=[agent_reviewer],
+            function=termination_function,
+            kernel=_create_kernel_with_chat_completion("termination"),
+            result_parser=lambda result: str(result.value[0]).lower() == "yes",
```

Comment:
Do we have an option to terminate when a given function is called, rather than running a function and parsing the output? For example, can I create a KernelFunction called `Terminate()` with a description of `Call this function when the goal has been achieved and the conversation should terminate` and then use that function call itself as the signal to terminate?

Reply:
As of right now, no. But let me look into it.